File: DSA/DeveshSrivastava_590017127/Day5/3.cpp
#include <iostream>
#include <vector>

using namespace std;

bool checkPalindrome(const vector<int> &nums) {
  int left = 0;
  int right = nums.size() - 1;
  while (left < right) {
    if (nums[left] != nums[right]) {
      return false;
    }
    left++;
    right--;
  }
  return true;
}

int main() {

  vector<int> nums = {4, 3, 2, 1, 2, 3, 4};
  checkPalindrome(nums) ? cout << "true" : cout << "false";
}

File: DSA/Sandeep_Jha_590014310/week2/day11/3.py
def count_palin_sub(s):
    n = len(s)
    count = 0
    def expand(left, right):
        nonlocal count
        while left >= 0 and right < n and s[left] == s[right]:
            count += 1
            left -= 1
            right += 1
    for i in range(n):
        expand(i, i)
        expand(i, i+1)
    return count

print(count_palin_sub(s=input("enter your word: ")))